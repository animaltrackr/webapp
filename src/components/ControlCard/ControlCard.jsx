import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Card, Icon } from 'antd';
import './ControlCard.less';

import DeerSelector from '../DeerSelector';
import DateRangeForm from '../DateRangeForm';

class ControlCard extends Component {
	render() {
		return (
			<Card className="control-card" bodyStyle={{ padding: '0px' }}>
				<div className="control-section">
					<div className="selector-row">
						<DeerSelector
							deerStates={this.props.deerStates}
							toggleDeer={this.props.toggleDeer}
							loading={this.props.loading}
						/>
						<Icon
							className="control-icon"
							type={`eye${this.props.showAll ? '-invisible' : ''}`}
							onClick={this.props.toggleAllDeer}
						/>
					</div>
					<div className="filter-row">
						<Icon
							className={`control-icon${
								this.props.dateFilterVisible ? '-active' : ''
							}`}
							type="calendar"
							onClick={this.props.toggleDateFilter}
						/>
					</div>
				</div>
				<div
					className={`secondary-section${
						this.props.dateFilterVisible ? '' : '-hidden'
					}`}
				>
					<Card
						className="secondary-card"
						bodyStyle={{ padding: '5px', margin: '5px' }}
					>
						<DateRangeForm
							handleDateFilter={this.props.handleDateFilter}
							handleDrawer={this.props.toggleDateFilter}
						/>
					</Card>
				</div>
			</Card>
		);
	}
}

ControlCard.propTypes = {
	deerStates: PropTypes.arrayOf(PropTypes.object).isRequired,
	showAll: PropTypes.bool.isRequired,
	dateFilterVisible: PropTypes.bool.isRequired,
	toggleDateFilter: PropTypes.func.isRequired,
	toggleDeer: PropTypes.func.isRequired,
	toggleAllDeer: PropTypes.func.isRequired,
	handleDateFilter: PropTypes.func.isRequired,
	loading: PropTypes.bool.isRequired,
};

export default ControlCard;
