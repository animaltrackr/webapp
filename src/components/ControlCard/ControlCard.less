.control-card {
	position: fixed;
	top: 60px;
	left: 10px;
	max-width: 400px;
	min-width: 250px;
	width: 25vw;
	box-shadow: 0px 1px 8px 0px rgba(0, 0, 0, 0.2),
		0px 3px 4px 0px rgba(0, 0, 0, 0.14), 0px 3px 3px -2px rgba(0, 0, 0, 0.12);
}

.control-section,
.secondary-section {
	padding-top: 10px;
	padding-left: 10px;
	padding-right: 10px;
	padding-bottom: 10px;
	& > * {
		margin-bottom: 5px;
	}
}

.secondary-section {
	background-color: #a7dd6936;
}
