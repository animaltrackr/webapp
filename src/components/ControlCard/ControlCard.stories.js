import React from 'react';
import { storiesOf } from '@storybook/react';
import ControlCard from './ControlCard';

storiesOf('ControlCard', module).add('ControlCard', () => (
	<div>
		<ControlCard />
	</div>
));
