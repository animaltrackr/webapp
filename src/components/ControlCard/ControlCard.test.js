import React from 'react';
import { shallow } from 'enzyme';
import ControlCard from './ControlCard';

describe('<ControlCard />', () => {
	test('renders', () => {
		const wrapper = shallow(<ControlCard />);
		expect(wrapper).toMatchSnapshot();
	});
});
