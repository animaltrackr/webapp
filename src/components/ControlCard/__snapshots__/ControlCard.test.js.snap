// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ControlCard /> renders 1`] = `ShallowWrapper {}`;
